### ISSUE TITLE
Replace with your own dataset

### ISSUE THREAD

**Issue by 135790zs** [bollossom/ICLR_TINY_SNN#1]
If I want to use my own image dataset, how can I modify this code? Thank you for your answer

**Comment by bollossom** [comment]
Change this 'Generation/datasets/load_dataset_snn.py' file and add your data loader

**Comment by 135790zs** [comment]
def load_dataset(data_path, batch_size, input_size):
    print("loading dataset")
    if not os.path.exists(data_path):
        raise ValueError("数据集路径不存在！")

    SetRange = transforms.Lambda(lambda X: 2 * X - 1.)
    transform = transforms.Compose([
        transforms.RandomHorizontalFlip(),
        transforms.CenterCrop(148),
        transforms.Resize((input_size, input_size)),
        transforms.ToTensor(),
        SetRange])

    trainset = CustomDataset(data_path, split='train', transform=transform)
    trainloader = DataLoader(trainset, batch_size=batch_size, shuffle=True, pin_memory=True)

    testset = CustomDataset(data_path, split='test', transform=transform)
    testloader = DataLoader(testset, batch_size=batch_size, shuffle=False, pin_memory=True)

    return trainloader, testloader


# 自定义Dataset类，用于加载自定义数据集
class CustomDataset(Dataset):
    def __init__(self, data_path, split='train', transform=None):
        self.data_path = data_path
        self.split = split
        self.transform = transform
        self.images = []  # 存储图片路径

        # 加载数据集
        self.load_dataset()

    def load_dataset(self):
        split_path = os.path.join(self.data_path, self.split)
        if not os.path.exists(split_path):
            raise ValueError(f"{self.split}数据集路径不存在！")

        # 获取split文件夹下的所有图片文件路径
        for image_file in os.listdir(split_path):
            image_path = os.path.join(split_path, image_file)
            self.images.append(image_path)

    def __len__(self):
        return len(self.images)

    def __getitem__(self, index):
        image_path = self.images[index]

        # 加载图像数据
        image = self.load_image(image_path)

        # 图像预处理
        if self.transform is not None:
            image = self.transform(image)

        return image

    @staticmethod
    def load_image(image_path):
        # 根据实际情况进行图像加载操作
        # 下面是一个示例，将图像加载为PIL.Image对象
        # 如果你的数据集使用其他格式，请根据实际情况进行修改
        from PIL import Image
        image = Image.open(image_path).convert('RGB')
        return image


# 定义数据集路径、批量大小和输入尺寸
data_path = r'H:\桌面\ASA-SNN-main\cwru28 115'
batch_size = 32
input_size = 64

# 调用数据加载器
trainloader, testloader = load_dataset(data_path, batch_size, input_size)

I'm a beginner, I modified it like this, but it doesn't work yet, can you give me some suggestions?

**Comment by 135790zs** [comment]
def load_dataset(data_path, batch_size, input_size):
    print("loading dataset")
    if not os.path.exists(data_path):
        raise ValueError("数据集路径不存在！")

    SetRange = transforms.Lambda(lambda X: 2 * X - 1.)
    transform = transforms.Compose([
        transforms.RandomHorizontalFlip(),
        transforms.CenterCrop(148),
        transforms.Resize((input_size, input_size)),
        transforms.ToTensor(),
        SetRange])

    trainset = CustomDataset(data_path, split='train', transform=transform)
    trainloader = DataLoader(trainset, batch_size=batch_size, shuffle=True, pin_memory=True)

    testset = CustomDataset(data_path, split='test', transform=transform)
    testloader = DataLoader(testset, batch_size=batch_size, shuffle=False, pin_memory=True)

    return trainloader, testloader


# 自定义Dataset类，用于加载自定义数据集
class CustomDataset(Dataset):
    def __init__(self, data_path, split='train', transform=None):
        self.data_path = data_path
        self.split = split
        self.transform = transform
        self.images = []  # 存储图片路径

        # 加载数据集
        self.load_dataset()

    def load_dataset(self):
        split_path = os.path.join(self.data_path, self.split)
        if not os.path.exists(split_path):
            raise ValueError(f"{self.split}数据集路径不存在！")

        # 获取split文件夹下的所有图片文件路径
        for image_file in os.listdir(split_path):
            image_path = os.path.join(split_path, image_file)
            self.images.append(image_path)

    def __len__(self):
        return len(self.images)

    def __getitem__(self, index):
        image_path = self.images[index]

        # 加载图像数据
        image = self.load_image(image_path)

        # 图像预处理
        if self.transform is not None:
            image = self.transform(image)

        return image

    @staticmethod
    def load_image(image_path):
        # 根据实际情况进行图像加载操作
        # 下面是一个示例，将图像加载为PIL.Image对象
        # 如果你的数据集使用其他格式，请根据实际情况进行修改
        from PIL import Image
        image = Image.open(image_path).convert('RGB')
        return image


# 定义数据集路径、批量大小和输入尺寸
data_path = r'H:\桌面\ASA-SNN-main\cwru28 115'
batch_size = 32
input_size = 64

# 调用数据加载器
trainloader, testloader = load_dataset(data_path, batch_size, input_size)

I'm a beginner, I modified it like this, but it doesn't work yet, can you give me some suggestions?
